app/lib/shared/exceptions.dart

```diff
@@ -157,14 +158,28 @@ class InvalidInputException extends ResponseException {
     _check(_ulidPattern.hasMatch(value), () => '"$name" is not a valid ulid.');
   }
 
-  static void checkSemanticVersion(String? version) {
+  /// Throw [InvalidInputException] if [package] is not an allowed package name.
+  static void checkPackageName(String? package) {
```

Comment:
We already have `checkPackageVersionParams` in `package/backend.dart`, it seems like a good candidate to merge this with.

Reply:
hmm, I used this in `checkPackageVersionParams`, but it allows  `latest` as version...